docker/pypi/wmagent/bin/manage

```diff
@@ -158,10 +163,69 @@ init_wmagent(){
 }
 
 
+##############################
+#  Tier0 stuff               #
+##############################
+
+
+# generate the tier0 config from the basic template
+init_tier0(){
+    local errVal=0
+    wmagent-mod-config --input=$WMA_CONFIG_DIR/config-template.py \
+                       --output=$WMA_CONFIG_DIR/config-agent.py \
+                       --coredb_url=oracle://$ORACLE_USER:$ORACLE_PASS@$ORACLE_TNS \
+                       --working_dir=$WMA_INSTALL_DIR \
+                       --couch_url=http://$COUCH_USER:$COUCH_PASS@$COUCH_HOST:$COUCH_PORT \
+                       --workqueue_db_name=$LOCAL_WORKQUEUE_DBNAME \
+                       --workload_summary_url=$WORKLOAD_SUMMARY_URL \
+                       --grafana_token=$GRAFANA_TOKEN \
+                       --wmstats_url=$WMSTATS_URL \
+                       --dbs3_url=$DBS3_URL \
+                       --dqm_url=$DQM_URL \
+                       --requestcouch_url=$REQUESTCOUCH_URL \
+                       --central_logdb_url=$CENTRAL_LOGDB_URL \
+                       --wmarchive_url=$WMARCHIVE_URL \
+                       --amq_credentials=$AMQ_CREDENTIALS \
+	                   --rucio_account=$RUCIO_ACCOUNT \
+	                   --rucio_host=$RUCIO_HOST \
+	                   --rucio_auth=$RUCIO_AUTH
+    let errVal+=$?
+
+    # NOTE: For the next step relates to modifying the originally generated WMAgent config file
+    #       from $WMA_CONFIG_DIR/config-agent.py
+    #       to $WMA_CONFIG_DIR/config.py
+    #       It removes the components not relevant for T0 and adds the additional DB credentials
+    #       used at Tier0Feeder. In order for it to be executed properly we need to download the
+    #       tier0-mod-config script from https://github.com/dmwm/T0/blob/master/bin/tier0-mod-config
+    optargs=""
+    [[ -n $CONFDB_URL ]]      && optargs="$optargs --confdb_url=$CONFDB_URL"
+    [[ -n $SMDB_URL ]]        && optargs="$optargs --smdb_url=$SMDB_URL"
+    [[ -n $POPCONLOGDB_URL ]] && optargs="$optargs --popconlogdb_url=$POPCONLOGDB_URL"
+    [[ -n $T0DATASVCDB_URL ]] && optargs="$optargs --t0datasvcdb_url=$T0DATASVCDB_URL"
+    [[ -n $SMNOTIFYDB_URL ]]  && optargs="$optargs --smnotifydb_url=$SMNOTIFYDB_URL"
+
+    tier0-mod-config --input=$WMA_CONFIG_DIR/config-agent.py \
+	                 --output=$WMA_CONFIG_DIR/config.py \
+                     $optargs
+    let errVal+=$?
+
+    wmcore-db-init --config $WMA_CONFIG_DIR/config.py --create --modules=WMCore.WMBS,WMCore.Agent.Database,WMComponent.DBS3Buffer,WMCore.BossAir,WMCore.ResourceControl,T0.WMBS;
+    let errVal+=$?
+    export WMAGENT_CONFIG=$WMA_CONFIG_DIR/config.py
+    wmagent-couchapp-init;
+    let errVal+=$?
+    unset WMAGENT_CONFIG
+}
+
+
 init_agent(){
     if _init_valid $wmaInitActive ; then
         echo "Initialising Agent..."
```

Comment:
Could this echo line be moved in the if-else statement, together with the specification of what type of agent is going to be initialized? Only to enforce the confidence that an operator has about deploying the right agent

Reply:
good point